test/DGmethods/compressible_Navier_Stokes/mms_bc.jl

```diff
@@ -23,176 +23,143 @@ else
-    τ12 = τ21 = VF[_τ12]
-    τ13 = τ31 = VF[_τ13]
-    τ23 = τ32 = VF[_τ23]
-
-    # inviscid terms
-    F[1, _ρ], F[2, _ρ], F[3, _ρ] = U          , V          , W
-    F[1, _U], F[2, _U], F[3, _U] = u * U  + P , v * U      , w * U
-    F[1, _V], F[2, _V], F[3, _V] = u * V      , v * V + P  , w * V
-    F[1, _W], F[2, _W], F[3, _W] = u * W      , v * W      , w * W + P
-    F[1, _E], F[2, _E], F[3, _E] = u * (E + P), v * (E + P), w * (E + P)
-
-    # viscous terms
-    F[1, _U] -= τ11; F[2, _U] -= τ12; F[3, _U] -= τ13
-    F[1, _V] -= τ21; F[2, _V] -= τ22; F[3, _V] -= τ23
-    F[1, _W] -= τ31; F[2, _W] -= τ32; F[3, _W] -= τ33
-
-    F[1, _E] -= u * τ11 + v * τ12 + w * τ13
-    F[2, _E] -= u * τ21 + v * τ22 + w * τ23
-    F[3, _E] -= u * τ31 + v * τ32 + w * τ33
-  end
+function wavespeed(::AtmosModel, nM, state::State, aux::State, t::Real)
+  γ = γ_exact
+  ρinv = 1 / state.ρ
+  u, v, w = ρinv * state.ρu, ρinv * state.ρv, ρinv * state.ρw
+  P = (γ-1)*(state.ρe - ρinv * (state.ρu^2 + state.ρv^2 + state.ρw^2) / 2)
+
+  return abs(nM[1] * u + nM[2] * v + nM[3] * w) + sqrt(ρinv * γ * P)
 end
 
-# Compute the velocity from the state
-@inline function velocities!(vel, Q, _...)
-  @inbounds begin
-    # ordering should match states_for_gradient_transform
-    ρ, U, V, W = Q[1], Q[2], Q[3], Q[4]
-    ρinv = 1 / ρ
-    vel[1], vel[2], vel[3] = ρinv * U, ρinv * V, ρinv * W
-  end
+function boundarycondition!(bl::AtmosModel, stateP::State, diffP::State, auxP::State, nM, stateM::State, diffM::State, auxM::State, bctype, t)
+  init_state!(bl, stateP, auxP, (auxM.x, auxM.y, auxM.z), t)
 end
 
-# Visous flux
-@inline function compute_stresses!(VF, grad_vel, _...)
-  μ::eltype(VF) = μ_exact
-  @inbounds begin
-    dudx, dudy, dudz = grad_vel[1, 1], grad_vel[2, 1], grad_vel[3, 1]
-    dvdx, dvdy, dvdz = grad_vel[1, 2], grad_vel[2, 2], grad_vel[3, 2]
-    dwdx, dwdy, dwdz = grad_vel[1, 3], grad_vel[2, 3], grad_vel[3, 3]
-
-    # strains
-    ϵ11 = dudx
-    ϵ22 = dvdy
-    ϵ33 = dwdz
-    ϵ12 = (dudy + dvdx) / 2
-    ϵ13 = (dudz + dwdx) / 2
-    ϵ23 = (dvdz + dwdy) / 2
-
-    # deviatoric stresses
-    VF[_τ11] = 2μ * (ϵ11 - (ϵ11 + ϵ22 + ϵ33) / 3)
-    VF[_τ22] = 2μ * (ϵ22 - (ϵ11 + ϵ22 + ϵ33) / 3)
-    VF[_τ33] = 2μ * (ϵ33 - (ϵ11 + ϵ22 + ϵ33) / 3)
-    VF[_τ12] = 2μ * ϵ12
-    VF[_τ13] = 2μ * ϵ13
-    VF[_τ23] = 2μ * ϵ23
-  end
+function init_aux!(::AtmosModel, aux::State, (x,y,z))
+  aux.x = x
+  aux.y = y
+  aux.z = z
+end
+
+function init_state!(bl::AtmosModel, state::State, aux::State, (x,y,z), t)
+  dim = dimension(bl)
+  state.ρ = ρ_g(t, x, y, z, Val(dim))
+  state.ρu = U_g(t, x, y, z, Val(dim))
+  state.ρv = V_g(t, x, y, z, Val(dim))
+  state.ρw = W_g(t, x, y, z, Val(dim))
+  state.ρe = E_g(t, x, y, z, Val(dim))
 end
 
-@inline function stresses_penalty!(VF, nM, velM, QM, aM, velP, QP, aP, t)
+import CLIMA.DGmethods.NumericalFluxes: GradNumericalFlux, diffusive_penalty!, diffusive_boundary_penalty!
+
+struct MyGradNumFlux <: GradNumericalFlux
```

Comment:
@lcw Do you have a better naming suggestion for this? Also, is what we've defined below a standard thing, should we move it to `src/DGmethods/NumericalFluxes.jl`?

Reply:
I don't have a better name.  @jkozdon and I have talked some about it and didn't come up with anything better.  The thing below is a standard central flux penalty.  And `diffusive_boundary_penalty!` is applying no boundary conditions.

Reply:
I think they could both be moved into `src/DGmethods/NumericalFluxes.jl`